dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserController.java

```diff
@@ -152,97 +149,81 @@ public class UserController extends AbstractCrudController<User> {
   // GET
   // -------------------------------------------------------------------------
 
-  @Override
-  @SuppressWarnings("unchecked")
-  protected List<User> getEntityList(
-      WebMetadata metadata,
-      WebOptions options,
-      List<String> filters,
-      List<Order> orders,
-      List<User> objects)
-      throws QueryParserException {
-    UserQueryParams params = makeUserQueryParams(options);
+  @Data
+  @EqualsAndHashCode(callSuper = true)
+  public static final class GetUserObjectListParams extends GetObjectListParams {
+    String phoneNumber;
+    boolean canManage;
+    boolean authSubset;
+    Date lastLogin;
+    Integer inactiveMonths;
+    Date inactiveSince;
+    boolean selfRegistered;
+    UserInvitationStatus invitationStatus;
+    boolean userOrgUnits;
+    boolean includeChildren;
+    UserOrgUnitType orgUnitBoundary;
+
+    @OpenApi.Property({UID.class, OrganisationUnit.class})
+    String ou;
+
+    boolean manage;
+
+    @JsonIgnore
+    boolean isUsingAnySpecialFilters() {
+      return phoneNumber != null
+          || canManage
+          || authSubset
+          || lastLogin != null
+          || inactiveMonths != null
+          || inactiveSince != null
+          || selfRegistered
+          || invitationStatus != null
+          || userOrgUnits
+          || includeChildren
+          || orgUnitBoundary != null
+          || ou != null
+          || manage;
+    }
+  }
 
-    String ou = options.get("ou");
+  @Override
+  protected List<UID> getPreQueryMatches(GetUserObjectListParams params) throws ConflictException {
+    if (!params.isUsingAnySpecialFilters()) return null;
+    UserQueryParams queryParams = toUserQueryParams(params);
 
+    String ou = params.getOu();
     if (ou != null) {
-      params.addOrganisationUnit(organisationUnitService.getOrganisationUnit(ou));
-    }
-
-    if (options.isManage()) {
-      params.setCanManage(true);
-      params.setAuthSubset(true);
+      queryParams.addOrganisationUnit(organisationUnitService.getOrganisationUnit(ou));
     }
 
-    boolean hasUserGroupFilter = filters.stream().anyMatch(f -> f.startsWith("userGroups."));
-    params.setPrefetchUserGroups(hasUserGroupFilter);
```

Comment:
FYI: This is no longer needed because the query will only result in a list of UIDs